extensions/amp-date-picker/1.0/single-date-picker.js

```diff
@@ -0,0 +1,142 @@
+import {DayPicker} from 'react-day-picker';
+
+import {
+  closestAncestorElementBySelector,
+  scopedQuerySelector,
+} from '#core/dom/query';
+
+import * as Preact from '#preact';
+import {useCallback, useEffect, useMemo, useRef, useState} from '#preact';
+import {ContainWrapper} from '#preact/component';
+
+import {
+  DateFieldNameByType,
+  DateFieldType,
+  DatePickerMode,
+  FORM_INPUT_SELECTOR,
+  TAG,
+} from './constants';
+import {getFormattedDate, parseDate} from './date-helpers';
+import {DayButton} from './day-button';
+
+/**
+ * @param {!DateInput.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function SingleDatePicker({
+  blockedDates,
+  children,
+  format,
+  id,
+  initialVisibleMonth,
+  inputSelector,
+  mode,
+  onError,
+  ...rest
+}) {
+  const [date, _setDate] = useState();
+  const [attributes, setAttributes] = useState();
+
+  const onErrorRef = useRef(onError);
+
+  const containerRef = useRef();
+
+  const [isOpen] = useState(mode === DatePickerMode.STATIC);
+
+  const [showChildren, setShowChildren] = useState(true);
+  /**
+   * Generate a name for a hidden input.
+   * Date pickers not in a form don't need named hidden inputs.
+   * @param {!Element} form
+   * @param {!DateFieldType} type
+   * @return {string}
+   * @private
+   */
+  const getHiddenInputId = useCallback(
```

Comment:
There's a little bit of duplication here between the single date picker and range picker, but I'm not sure whether I care enough to extract it somewhere since this function depends on the component's parent form element and error handling.